---
title: "R Notebook"
output: html_notebook
---


```{r install DESeq2}
source("https://bioconductor.org/biocLite.R")
#biocLite("DESeq2")
biocLite("sRAP")
install.packages("Rcpp")
install.packages("stringi")
```

```{r load in data}
library(data.table)

datCts <- data.frame(fread("RNAseq_WT_DAT.csv", select= c(7:29), skip = 1, header = TRUE), row.names= 1)
dat.Cts <- t(datCts)
dat.Cts <- data.frame(as.matrix(dat.Cts, row.names = FALSE))



datAnno <- file.path("dat_sample_annotation.txt")
                      #row.names= 1)

dat.Anno <- read.table("dat_sample_annotation.txt")



#for DeSeq2
##dat.exp.values <- write.table(datCts, "dat_exp_values.txt", sep = " ", row.names = TRUE, col.names = TRUE)
##DAT.expression <- read.table("dat_exp_values.txt")


#dat_cts <- as.matrix(datCts, rownames = "Transcript")
#dat_coldata <- datAnno[,c("drug_condit", "genotype"),]

```

```{r verify that all rownames appear in cts matrix: DeSeq2}
all(rownames(dat_coldata) %in% colnames(dat_cts))
```
```{r verify all rownames and colnames are equal: DeSeq2}
all(rownames(dat_coldata) == colnames(dat_cts))
```
```{r unnecessary?: Deseq2}
dat_cts <- dat_cts[, rownames(dat_coldata)]
all(rownames(dat_coldata) == colnames(dat_cts))
```

```{r}
library(Rcpp)
library(stringi)
library(sRAP)


project.folder <- getwd()
project.name <- "DAT_RNAseq"
```
```{r}

RNA.qc(datAnno, dat.Cts, project.name,
project.folder, plot.legend= FALSE,
color.palette = c("green", "orange"))
```


































































```{r}
library(assertthat)
library(magrittr)

infer.counts <- function(x, digits=3) {
    assert_that(all(x >= 0))
    assert_that(digits >= 2)
    ## Get all diffs between successive unique values
    diffs <- x %>% sort %>% unique %>% diff
    ## Round to a few digits to work around inexact representation
    approxdiffs <- signif(diffs, digits)
    ## Find the rounded interval that occurs most often
    approxguess <- approxdiffs %>% table %>% .[which.max(.)] %>% names %>% as.numeric
    ## Find all the intervals that were rounded to the selected one, and take their mean
    unit.guess <- diffs[approxdiffs == approxguess] %>% mean
    message("Guessing 1 count = ", unit.guess)
    ## Divide the original vector by the unit guess, and round to a
    ## few significant digits, which should ideally round everything
    ## to integers.
    round(x / unit.guess, digits)
}

```
```{r}
normcounts <- dat_cts
counts <- apply(normcounts, 2, infer.counts)
```
```{r}
normcounts[1:5, 1:5]
```
```{r}
counts[1:5, 1:5]
```







```{r}
library("DESeq2")
dat_dds <- DESeqDataSetFromMatrix(countData = dat_cts,
                              colData = dat_coldata,
                              design = ~ drug_condit)
dat_dds
```

